app/assets/javascripts/event_staff.js

```diff
@@ -1,23 +1,56 @@
-$(document).ready(function(){
-  $(".add_staff_field").autocomplete({
-    delay: 500,
-    minLength: 4,
-    source: "/lookup_staffs",
-    select: function(event, ui){
-      var staffId = ui.item.data;
-      if ($(".event_staffs").length > 0){
-        $.get("/user_info/" + staffId, function(data){
-          $(".event_staffs").append(data);
-          $(".add_staff_field").val("");
-        });
-      }
-      else {
-          $("#new_staff").val(staffId);
-      }
-    }
-  });
 
-  $(".event_staffs").on("click", ".remove_staff", function(){
-    $(this).parents(".chip").hide("slow", function() { $(this).remove(); });
+  $(document).ready(function(){
+
+    var fromPage = document.getElementsByClassName("chip");
+    $(".add_staff_field").autocomplete({
+        delay:500,
+        minLength: 4,
+        source: "/lookup_staffs",
+        select:function(event, ui){
+          var staffId = ui.item.data;
+          $("#staff_id").val(staffId);
+        }
+    });
+
+
+      $(".event_staffs").on("click", ".remove_staff", function(){
+        $(this).parents(".chip").hide("slow", function() { $(this).remove(); });
+      });
+
+
+      $("#add_staff").on("click", function(){
+        var checkField = $("#staff_field_id").val();
+
+        if (checkField !== ""){
+          var staffId = $("#staff_id").val();
+          var staffRole = $("[name='role']").val();
+          $.get("/user_info/" +staffId +"?role=" + staffRole, function(data){
+            if(data.indexOf("<html>") > 0){
```

Comment:
What do you guys think of refactoring line 28 - 43 into a function of its own. This is because `#add_staff` click eventhandler function is too clustered with code.